src/overlays/actors/ovl_En_Torch2/z_en_torch2.c

```diff
@@ -19,18 +31,137 @@ const ActorInit En_Torch2_InitVars = {
     (ActorFunc)EnTorch2_Init,
     (ActorFunc)EnTorch2_Destroy,
     (ActorFunc)EnTorch2_Update,
-    (ActorFunc)EnTorch2_Draw
+    (ActorFunc)EnTorch2_Draw,
+};
+
+static ColliderCylinderInit sCylinderInit = {
+    { COLTYPE_METAL, AT_NONE, AC_ON | AC_HARD | AC_TYPE_PLAYER, OC1_ON | OC1_TYPE_PLAYER | OC1_TYPE_1 | OC1_TYPE_2,
+      OC2_TYPE_2, COLSHAPE_CYLINDER },
+    { ELEMTYPE_UNK2, { 0x00100000, 0, 0 }, { 0xF7CFFFFF, 0, 0 }, TOUCH_NONE, BUMP_ON | BUMP_HOOKABLE, OCELEM_ON },
+    { 20, 60, 0, { 0, 0, 0 } },
+};
+
+static InitChainEntry sInitChain[] = {
+    ICHAIN_S8(colChkInfo.mass, MASS_IMMOVABLE, ICHAIN_STOP),
+};
+
+// Statues for each of Link's different forms
+// (Playing elegy as Fierce Deity puts down a human statue)
+static Gfx* const sGfxs[] = {
+    &D_0401C430, &D_04048DF0, &D_04089070, &D_04057B10, &D_0401C430,
 };
-*/
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Torch2_0x808A31B0/EnTorch2_Init.asm")
+void EnTorch2_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnTorch2* this = THIS;
+    s16 params;
+
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    Collider_InitAndSetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+
+    // params: which form Link is in (e.g. human, deku, etc.)
+    params = this->actor.params;
+    if (params != 3) {
+        this->actor.flags |= 0x4000000;
+        if (params == 1) {
+            this->actor.flags |= 0x20000;
+        }
+    }
+    this->framesUntilNextStep = 20;
+}
+
+void EnTorch2_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnTorch2* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+    func_80169DCC(globalCtx, this->actor.params + 3, 0xFF, 0, 0xBFF, &this->actor.world, this->actor.shape.rot.y);
+    globalCtx->actorCtx.unk254[this->actor.params] = 0;
+}
+
+void EnTorch2_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnTorch2* this = THIS;
+
+    u16 targetAlpha;
+    u16 remainingFrames;
+    u32 unused0;
+    u32 unused1;
+
+    if (this->step == 3) {
+        this->actor.update = &EnTorch2_UpdateIdle;
```

Comment:
The `&` should be unnecessary. I don't recall if IDO warns but GCC should warn about this. In OoT we have a CC_CHECK step where files are ran through GCC to test warnings and syntax of `NON_MATCHING`s. This isn't in MM right now but hopefully it will be in the future.
Also applies to the other update assignment.